src/pages/membership/components/createYouthProfile/__tests__/CreateYouthProfile.test.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { MockedProvider, MockedResponse } from '@apollo/react-testing';
+import { User } from 'oidc-client';
+import { loader } from 'graphql.macro';
+
+import i18n from '../../../../../common/test/testi18nInit';
+import CreateYouthProfile from '../CreateYouthProfile';
+import { updateWrapper } from '../../../../../common/test/testUtils';
+import {
+  Language,
+  PrefillRegistartion,
+} from '../../../../../graphql/generatedTypes';
+
+const PREFILL_REGISTRATION = loader(
+  '../../../graphql/PrefillRegistration.graphql'
+);
+
+/* eslint-disable @typescript-eslint/camelcase */
+const tunnistamoUser: User = {
+  profile: {
+    iss: '',
+    sub: '',
+    aud: '',
+    exp: 0,
+    iat: 0,
+  },
+  access_token: '',
+  expired: false,
+  expires_at: 0,
+  expires_in: 0,
+  id_token: '',
+  scope: '',
+  scopes: [],
+  state: undefined,
+  token_type: '',
+  toStorageString(): string {
+    return '';
+  },
+};
+/* eslint-enable @typescript-eslint/camelcase */
+
+type MyProfile = {
+  language?: string;
+};
+
+const getMocks = (myProfile: MyProfile) => {
+  return [
+    {
+      request: {
+        query: PREFILL_REGISTRATION,
+        variables: {},
+      },
+      result: {
+        loading: false,
+        data: {
+          myProfile: {
+            firstName: 'Teemu',
+            lastName: 'Testaaja',
+            language: Language.FINNISH,
+            ...myProfile,
+            primaryAddress: {
+              address: 'Testikuja 55',
+              postalCode: '00100',
+              city: 'Helsinki',
+              countryCode: 'FI',
+              id: '123',
+              __typename: 'AddressNode',
+            },
+            primaryEmail: {
+              email: 'teemu.testaaja@test.fi',
+              __typename: 'EmailNode',
+            },
+            primaryPhone: {
+              id: '123',
+              phone: '05501234567',
+              __typename: 'PhoneNode',
+            },
+            __typename: 'ProfileNode',
+          },
+        } as PrefillRegistartion,
+      },
+    },
+  ];
+};
+
+const getWrapper = (mocks?: MockedResponse[]) => {
+  return mount(
+    <MockedProvider mocks={mocks}>
```

Comment:
You could setup this provider here: https://github.com/City-of-Helsinki/youth-membership-ui/blob/develop/src/common/test/testUtils.tsx

As far as I know it's a common pattern for projects with Apollo.

Reply:
Added `mountWithApolloProvider` to utilities.